pkg/task/queue.go

```diff
@@ -90,6 +90,34 @@ func (q *Queue) Pop() (*Task, error) {
 	return tsk, nil
 }
 
+// Remove all existing tasks from the queue that match the given branch/string
+func (q *Queue) RemoveExisting(branch string, repo string) error {
+	q.Lock()
+	defer q.Unlock()
+
+	var err error
+	keep_indexes := make([]int, 0)
+	for index, task := range *q.tq {
+		// if task matches both branch and repo, delete it
+		if task.CreatedBy.Repo == repo && task.CreatedBy.Branch == branch {
+			err = q.ts.Delete(task.ID)
+			if err != nil {
+				return err
+			}
+		} else {
+			keep_indexes = append(keep_indexes, index)
+		}
+	}
+	keep_tasks := make([]*Task, len(keep_indexes))
+	for index, value := range keep_indexes {
+		keep_tasks[index] = (*q.tq)[value]
+	}
+
+	*q.tq = keep_tasks
+
+	return nil
+}
+
```

Comment:
It looks like we're dropping tasks, are we marking them as canceled?

Reply:
I wanted to remove the tasks from the DB  (ie. mark them as cancelled), but I wasn't sure what's the proper way to handle this.
Should I use `storage.Delete(taskId)`, or `ArchiveTask(task)` ?

Reply:
I'm not sure but this looks storage related,
the way to go is probably to check the `supervisor.go` and see what happens when a task is canceled, I know we have states `StateCanceled` and `OutcomeCanceled` and I expect to see these, but I don't know the exact workflow atm.